src/java.base/unix/native/libnio/fs/UnixCopyFile.c

```diff
@@ -37,6 +37,10 @@
 #endif
 #include "sun_nio_fs_UnixCopyFile.h"
 
+#if ! defined(_ALLBSD_SOURCE)
```

Comment:
The "#if  ! defined(_ALLBSD_SOURCE)" looks odd here. Maybe we can get rid of it, it doesn't matter if this tiny function is compiled on macOS.

Reply:
OK

Reply:
I moved `transfer()` so it is complied on all platforms, but it could cleanly be in an `#else` branch of the conditional compilation

```
#if defined(_ALLBSD_SOURCE)
int fcopyfile_callback(int what, int stage, copyfile_state_t state,
    const char* src, const char* dst, void* cancel)
```